generic_config_updater/generic_updater.py

```diff
@@ -67,8 +67,12 @@ def apply(self, patch):
             self.logger.log_notice(f"  * {change}")
 
         # Apply changes in order
-        self.logger.log_notice("Applying changes in order.")
+        if changes_len:
+            self.logger.log_notice(f"Applying {changes_len} change{'s' if changes_len != 1 else ''} in order:")
+        else:
+            self.logger.log_notice(f"No changes to apply.")
```

Comment:
To make it simple, how about
```
            self.logger.log_notice(f"Applying {changes_len} change(s) in order:")
```

And no need to use if-else.

Reply:
I prefer log messages to be very informative even if they need a little extra code.

In case of 0, I prefer printing a clear message saying
```
No changes to apply.
```

In case of 1 change:
```
Applying 1 change in order:
```

In case of many changes:
```
Applying x changes in order:
```

I think this makes the logs easier to read.

Reply:
I can make it
```
self.logger.log_notice(f"Applying {changes_len} change{'s' if changes_len != 1 else ''} in order{':' if changes_len > 0 else '.'}")
```

Reply:
updated